### PR TITLE
Breadcrumb rework

### LINKED ISSUE #2389: Visuelle Unterscheidung im Breadcrumb zwischen Ordnungssystem und Dossier
Dies wurde am Innovation Day 2016 als Verbesserungswunsch eingebracht. Man könnte aus meiner Sicht das Icon der Inhalte im Breadcrumb mit einblenden.

### PR DESCRIPTION
**Pull request by phgross**
bildschirmfoto 2017-08-23 um 07 33 48

bildschirmfoto 2017-08-23 um 07 33 27

This PR replaces the current implementation of the Pathbar viewlet, with our own to provide the following features:
 - extend the crumbs with icons
 - group the repository in to a dropdown list, to save space

It also no longer uses the breadcrumb view and uses the aq chain to generate the breadcrumb.

⚠️ Needs styling PR (https://github.com/4teamwork/plonetheme.teamraum/pull/573) merge first

Closes #2389

### CODE REVIEW

File: opengever/base/browser/helper.py [hunk 1]
```diff
@@ -10,13 +10,22 @@ def _get_task_css_class(task):
...
     globalindex object.
     """
-
     if ICatalogBrain.providedBy(task):
         task = Task.query.by_brain(task)
 
     return task.get_css_class()
 
 
+def get_contenttype_class(obj):
```
Comment (deiferni) [inline opengever/base/browser/helper.py]: this function should be called from `get_contenttype_class` then

Comment (deiferni) [inline opengever/base/browser/helper.py]: hm or maybe don't make it a separate function but use a boolean parameter in `get_css_class` to indicate whether we want the detailed content-icon or the object-type icon?

File: opengever/base/browser/helper.py [hunk 2]
```diff
@@ -10,13 +10,22 @@ def _get_task_css_class(task):
...
+    """Returns the css class corresponding the contenttype of the given object.
```
Comment (deiferni) [inline opengever/base/browser/helper.py]: to the

File: opengever/base/browser/helper.py [hunk 3]
```diff
@@ -10,13 +10,22 @@ def _get_task_css_class(task):
...
+
+    In differnt of `get_css_class` it does not handle document mimetypes or
```
Comment (deiferni) [inline opengever/base/browser/helper.py]: In differnt of -> Unlike

File: opengever/base/tests/test_pathbar.py [hunk 4]
```diff
@@ -14,20 +14,46 @@ def setUp(self):
         self.repo = create(Builder('repository')
                            .within(self.root)
                            .titled(u'Testposition'))
-        self.dossier = create(Builder('dossier')
+        self.subrepo = create(Builder('repository')
```
Comment (deiferni) [inline opengever/base/tests/test_pathbar.py]: could you reimplement these as `IntegrationTestCase`? should be a quick win 😉 .

File: opengever/base/viewlets/pathbar.py [hunk 5]
```diff
@@ -1,20 +1,64 @@
...
+        repository = []
+        chain = []
+        for obj in aq_chain(self.context):
+            if INavigationRoot.providedBy(obj):
+                break
+
+            if ISQLObjectWrapper.providedBy(obj):
+                data = obj.get_breadcrumb()
+
+            elif IHideFromBreadcrumbs.providedBy(obj):
```
Comment (deiferni) [inline opengever/base/viewlets/pathbar.py]: i would move `IHideFromBreadcrumbs` up by one if block

File: opengever/base/viewlets/pathbar.py [hunk 6]
```diff
@@ -1,20 +1,64 @@
...
+from opengever.repository.interfaces import IRepositoryFolder
+from opengever.repository.repositoryroot import IRepositoryRoot
+from plone.app.layout.navigation.interfaces import INavigationRoot
 from plone.app.layout.viewlets import common
+from Products.CMFPlone.interfaces import IHideFromBreadcrumbs
 from Products.Five.browser.pagetemplatefile import ViewPageTemplateFile
 
 
 class PathBar(common.PathBarViewlet):
+    """The breadcrumb viewlet.
```
Comment (deiferni) [inline opengever/base/viewlets/pathbar.py]: you could add a short description about what it does, e.g:

Groups repository objects and displays them in a dropdown.

File: opengever/base/viewlets/pathbar.py [hunk 7]
```diff
@@ -1,20 +1,64 @@
...
-        super(PathBar, self).update()
 
-        if getattr(self.view, 'has_model_breadcrumbs', False):
-            self.append_model_breadcrumbs()
+        repository_items, self.obj_chain = self.get_chains()
+
+        if repository_items:
+            self.leaf_node = repository_items[0]
+            if len(repository_items) > 1:
+                self.repository_chain = repository_items[1:]
```
Comment (deiferni) [inline opengever/base/viewlets/pathbar.py]: could this code be moved to `__init__`? or does initialisation happen too early?

File: opengever/base/viewlets/pathbar.py [hunk 8]
```diff
@@ -1,20 +1,64 @@
...
+    repository_chain = None
+    obj_chain = None
+
     def admin_unit_label(self):
         return get_current_admin_unit().label()
 
     def update(self):
-        super(PathBar, self).update()
 
-        if getattr(self.view, 'has_model_breadcrumbs', False):
```
Comment (deiferni) [inline opengever/base/viewlets/pathbar.py]: please also remove the attribute from all views, it should no longer be necessary

File: opengever/base/viewlets/pathbar.py (hunk 7 above)
Comment (phgross) [inline opengever/base/viewlets/pathbar.py]: Thats how the default `PathBarViewlet` and all other `plone.app.layout.viewlets` are  implemented. So i would prefer to not change it ...

**Comment by phgross** [comment]
@deiferni updated the PR according to your feedback and also converted the the pathbar tests to integration tests. Please habe another look ...

**Pull request merged**